llm/ggml.go

```diff
@@ -37,9 +38,47 @@ const (
 	FileTypeQ4_K
 	FileTypeQ5_K
 	FileTypeQ6_K
-	FileTypeUnknown = -1
 )
 
+func ParseFileType(family ModelFamily, s string) (FileType, error) {
+	switch family {
+	case ModelFamilyLlama:
+		return llamaParseFileType(s)
+	}
+
+	// if there's no family specific file type, parse as generic ggml file type
+	switch strings.ToLower(s) {
+	case "f32":
```

Comment:
Just verifying, there's no way we can infer this from the actual file data rather than the name?

Reply:
The way it's used right now, it's a helper function parsing the user input file type into the actual file type, such as defining what to quantize the model into.